typescript/src/services/redemptions/redemptions-service.ts

```diff
@@ -51,6 +52,73 @@ export class RedemptionsService {
   ): Promise<{
     targetChainTxHash: Hex
     walletPublicKey: Hex
+  }> {
+    const { walletPublicKey, mainUtxo, redeemerOutputScript } =
+      await this.determineRedemptionData(bitcoinRedeemerAddress, amount)
+
+    const txHash = await this.tbtcContracts.tbtcToken.requestRedemption(
+      walletPublicKey,
+      mainUtxo,
+      redeemerOutputScript,
+      amount
+    )
+
+    return {
+      targetChainTxHash: txHash,
+      walletPublicKey,
+    }
+  }
+
+  /**
+   * Prepare tBTC Redemption Data in the raw bytes format expected by the tBTC
+   * Bridge contract. The data is used to request a redemption of TBTC v2 token
+   * through custom integration with the tBTC Bridge contract.
+   * @param bitcoinRedeemerAddress Bitcoin address redeemed BTC should be
+   *                               sent to. Only P2PKH, P2WPKH, P2SH, and P2WSH
+   *                               address types are supported.
+   * @param amount The amount to be redeemed with the precision of the tBTC
+   *               on-chain token contract.
+   * @param chainRedeemerAddress Chain identifier of the redeemer. This is the
+   *                             address that will be able to claim the tBTC tokens
+   *                             if anything goes wrong during the redemption process.
+   * @returns Data needed to request a redemption.
+   */
+  async buildRedemptionData(
```

Comment:
What do you think about making this function more high-level. Something like that:
```
async requestRedemptionWithProxy(
    bitcoinRedeemerAddress: string,
    amount: BigNumber,
    redeemerProxy: RedeemerProxy
): Promise<{
    targetChainTxHash: Hex
    walletPublicKey: Hex
}> {
    const chainRedeemerAddress = redeemerProxy.redeemerAddress()

    const { walletPublicKey, mainUtxo, redeemerOutputScript } =
      await this.determineRedemptionData(bitcoinRedeemerAddress, amount)

    const redemptionData = this.tbtcContracts.tbtcToken.buildRequestRedemptionData(
      chainRedeemerAddress,
      walletPublicKey,
      mainUtxo,
      redeemerOutputScript
    )

    return redeemerProxy.requestRedemption(redemptionData) // this calls approveAndCall underneath
}

interface RedeemerProxy {
    redeemerAddress(): string

    requestRedemption(redemptionData: Hex): Promise<{
        targetChainTxHash: Hex
        walletPublicKey: Hex
    }>
}
```

That way we make tBTC SDK API consistent with what we have on the deposit side. Moreover, we can expose some base classes implementing `RedeemerProxy` to handle some redemption complexity and do a favor for integrators.

Reply:
As you wish 😉  I pushed the changes in 97ea592b02021439989ff06645c91ef343fa827f